src/transformers/models/resnet/modeling_resnet.py

```diff
@@ -0,0 +1,435 @@
+        self.pool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
+
+
+class ResNetShortCut(nn.Sequential):
+    """
+    ResNet shortcut, used to project the residual features to the correct size. If needed, it is also used to
+    downsample the input using `stride=2`.
+    """
+
+    def __init__(self, in_channels: int, out_channels: int, stride: int = 2):
+        super().__init__()
+        self.conv = nn.Conv2d(in_channels, out_channels, kernel_size=1, stride=stride, bias=False)
+        self.bn = nn.BatchNorm2d(out_channels)
+
+
+class ResNetBasicLayer(nn.Module):
+    """
+    A classic ResNet's residual layer composed by a two `3x3` convolutions.
+
+    Args:
+        in_channels (`int`):
+            The number of input channels.
+        out_channels (`int`):
+            The number of outputs channels.
+        stride (`int`, *optional*, defaults to 1):
+            The stride used in the first convolution.
+        activation (`int`, *optional*, defaults to `"relu"`):
+            The activation used by the layer.
+    """
+
+    def __init__(self, in_channels: int, out_channels: int, stride: int = 1, activation: str = "relu"):
+        super().__init__()
+        should_apply_shortcut = in_channels != out_channels or stride != 1
+        self.shortcut = (
+            ResNetShortCut(in_channels, out_channels, stride=stride) if should_apply_shortcut else nn.Identity()
+        )
+        self.layer = nn.Sequential(
+            ResNetConvLayer(in_channels, out_channels, stride=stride),
+            ResNetConvLayer(out_channels, out_channels, activation=None),
+        )
+        self.activation = ACT2FN[activation]
+
+    def forward(self, pixel_values):
+        residual = pixel_values
+        pixel_values = self.layer(pixel_values)
+        residual = self.shortcut(residual)
+        pixel_values += residual
+        pixel_values = self.activation(pixel_values)
+        return pixel_values
+
+
+class ResNetBottleNeckLayer(nn.Module):
+    """
+
+    A classic ResNet's bottleneck layer composed by a three `3x3` convolutions.
+
+    The first `1x1` convolution reduces the input by a factor of `reduction` in order to make the second `3x3`
+    convolution faster. The last `1x1` convolution remap the reduced features to `out_channels`.
+
+
+    Args:
+        in_channels (`int`):
+            The number of input channels.
+        out_channels (`int`):
+            The number of outputs channels.
+        stride (`int`, *optional*, defaults to 1):
+            The stride used in the first convolution.
+        activation (`int`, *optional*, defaults to `"relu"`):
+            The activation used by the layer.
+        reduction (`int`, *optional*, defaults to 4):
+            The reduction factor the block applies in the first `1x1` convolution.
+    """
+
+    def __init__(
+        self, in_channels: int, out_channels: int, stride: int = 1, activation: str = "relu", reduction: int = 4
+    ):
```

Comment:
Same here, there's already a great docstring, not sure whether these type annotations are useful.

Reply:
Comment still open.

Reply:
I think I can remove the parameters docstring in non important docstring

Reply:
Ok, feel free to remove.